modules/standard/CTypes.chpl

```diff
@@ -727,42 +725,145 @@ module CTypes {
     return c_pointer_return_const(arr[arr.domain.low]);
   }
 
+  // TODO: make the string and bytes versions of c_ptrTo and c_ptrToConst
+  // return a `c_ptr(c_char)` and `c_ptrConst(c_char)` respectively once
+  // c_string is deprecated and `c_ptr(c_char)` is supported.
+
+  /*
+    Returns a :type:`c_ptr` to the underlying buffer of a :type:`~String.string`
+
+    Note that the existence of this ``c_ptr`` has no impact on the lifetime of
+    the ``string``.  The returned pointer will be invalid if the ``string`` is
+    freed or even reallocated.
+
+    Halts if the ``string`` is empty and bounds checking is enabled.
+  */
+  inline proc c_ptrTo(ref s: string): c_ptrConst(c_uchar) {
```

Comment:
Per offline conversation I believe we should return a non-const `c_ptr` for `c_ptrTo` on string or bytes.